58-poodle-moth/technical_specification.md

```diff
@@ -0,0 +1,48 @@
+# Storing Data as strings vs BSON in MongoDB (Poodle Moth)
+**Epic Type:** Exploratory Epic
+
+Epic planning and implementation follows the
+[Epic Planning and Marathon SOP](https://docs.ghga-dev.de/main/sops/sop001_epic_planning.html)
+
+## Principle Components of Exploration:
+
+We currently store data in MongoDB with a process that converts more complex data types,
+such as datetimes, paths, and UUID objects, as their string representations.
+More precisely, we perform `<model>.model_dump_json()` on the given pydantic model
+before insertion. When the data is retrieved from the database, pydantic is able to
+reconstruct the original model, e.g. a stringified UUID is converted to an actual UUID.
+
+The above process is straightforward and uniform in its application, but introduces
+drawbacks in the form of query complexity and reduced efficiency. MongoDB stores data
+internally in a format called BSON (Binary JSON), which features support for many types
+of data, like datetimes and UUIDs. Instead of storing the string representations of
+such types, we could be storing them in, and querying with, their native format.
+
+This epic will explore the tradeoffs between storing data as we do now (string format)
+and migrating to BSON-supported formats. 
+
+Results to be produced:
+- ADR
+- Implementation Epic & tasks, only if BSON format is elected
+
+- \<Provide a list here.\>
+
+## Not part of this Exploration:
+
+All aspects that shall be ignored during the exploration:
+
+- Actual implementation of any changes deemed necessary
+
+## Material and Resources:
+
+Materials to be studied (e.g. articles, book chapters, youtube videos, etc.) and any (external) persons/experts to talk to:
+
+- [MongoDB Documentation](https://www.mongodb.com/resources/basics/json-and-bson)
```

Comment:
particularly: [MongoDB: JSON and BSON](https://www.mongodb.com/resources/basics/json-and-bson)